UNIXFS.md

```diff
@@ -59,6 +76,31 @@ For files that are comprised of more than a single block, the 'Type' field will
 
 This data is serialized and placed inside the 'Data' field of the outer merkledag protobuf, which also contains the actual links to the child nodes of this object.
 
+For files comprised of a single block, the 'Type' field will be set to 'File', 'filesize' will be set to the total number of bytes in the file and the file data will be stored in the 'Data' field.
+
+The serialized size of a UnixFS node must not exceed 256KiB in order to work will with the [Bitswap] protocol.
+
+## Metadata
+
+UnixFS currently supports two optional metadata fields:
+
+* `mode` -- The `mode` is for persisting the [file permissions in numeric notation](https://en.wikipedia.org/wiki/File_system_permissions#Numeric_notation) \[[spec](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html)\]. If unspecified this defaults to `0755` for directories/HAMT shards and `0644` for all other types where applicable.
+* `mtime` -- The modification time in seconds since the epoch. This defaults to the unix epoch if unspecified.
+
+### Inheritance
+
+When traversing down through a UnixFSv1.5 directory, child entries without metadata fields will inherit those of their direct ascendants.
```

Comment:
Is this really what we want?

Reply:
I think so?  Maybe the wording is wrong - you can have perms set at the root of a tree, which will cascade to all descendants, then you can specialise sections of the tree by setting perms on intermediate directories, and/or you can set perms on individual files for granular control.  You then save space by not having to set the perms on every node.

Reply:
If you copied a file that had inherited perms from one UnixFS directory into another, presumably you'd want the option of preserving those perms, so you'd resolve them from the parent and apply them to the resulting node which would change its CID.  I think that's ok.  Bit weird, but ok.

Reply:
This has the same problems as putting the permissions inside directories instead of inside files: if I give you a CID, it'll have different permissions/timestamps.

Reply:
I had the same thought while travelling today. I’ll remove this section.